tests/lib/cli-engine/cli-engine.js

```diff
@@ -3709,46 +3898,66 @@ describe("CLIEngine", () => {
         describe("don't ignore the entry directory.", () => {
             const root = getFixturePath("cli-engine/dont-ignore-entry-dir");
 
-            it("'executeOnFiles(\".\")' should not load config files from outside of \".\".", () => {
-                CLIEngine = defineCLIEngineWithInMemoryFileSystem({
-                    cwd: () => root,
+            /* eslint-disable-next-line func-style */
+            let cleanup = () => { };
+
+            afterEach(async () => {
+                await cleanup();
+                cleanup = () => { };
+            });
+
+            it("'executeOnFiles(\".\")' should not load config files from outside of \".\".", async () => {
+                const teardown = createCustomTeardown({
+                    cwd: root,
                     files: {
                         "../.eslintrc.json": "BROKEN FILE",
                         ".eslintrc.json": JSON.stringify({ root: true }),
                         "index.js": "console.log(\"hello\")"
                     }
-                }).CLIEngine;
-                engine = new CLIEngine();
+                });
+
+                engine = new CLIEngine({ cwd: teardown.getPath() });
+
+                await teardown.prepare();
+                cleanup = teardown.cleanup;
 
                 // Don't throw "failed to load config file" error.
                 engine.executeOnFiles(".");
             });
 
-            it("'executeOnFiles(\".\")' should not ignore '.' even if 'ignorePatterns' contains it.", () => {
-                CLIEngine = defineCLIEngineWithInMemoryFileSystem({
-                    cwd: () => root,
+            it("'executeOnFiles(\".\")' should not ignore '.' even if 'ignorePatterns' contains it.", async () => {
```

Comment:
```suggestion
            it.skip("'executeOnFiles(\".\")' should not ignore '.' even if 'ignorePatterns' contains it.", async () => {
```

I'm also seeing that the broken `../.eslintrc.json` file isn't cleaned up at the end of the previous test as @mdjermanovic found. It's only cleaned up when the whole temp fixture directory is deleted at the very end. If you add a `.skip` to this test, all the remaining tests that don't specify `root: true` will fail.

One solution could be to add another level of directory nesting to `root` for the two `../.eslintrc.json` tests so that the outermost `.eslintrc.json` is still inside the teardown directory.

Reply:
I really don't want to change how the current tests operate, as they are already complicated enough. I think the easiest solution is just to remove the extra file as part of the cleanup.